src/Database/Expression/CrossSchemaTableExpression.php

```diff
@@ -0,0 +1,95 @@
+<?php
+/**
+ * CakePHP(tm) : Rapid Development Framework (http://cakephp.org)
+ * Copyright (c) Cake Software Foundation, Inc. (http://cakefoundation.org)
+ *
+ * Licensed under The MIT License
+ * For full copyright and license information, please see the LICENSE.txt
+ * Redistributions of files must retain the above copyright notice.
+ *
+ * @copyright     Copyright (c) Cake Software Foundation, Inc. (http://cakefoundation.org)
+ * @link          http://cakephp.org CakePHP(tm) Project
+ * @since         3.3.0
+ * @license       http://www.opensource.org/licenses/mit-license.php MIT License
+ */
+namespace Cake\Database\Expression;
+
+use Cake\Database\ExpressionInterface;
+use Cake\Database\ValueBinder;
+
+/**
+ * An expression object that represents a cross schema table name
+ *
+ * @internal
+ */
+class CrossSchemaTableExpression implements ExpressionInterface
+{
+
+    /**
+     * Name of the schema
+     *
+     * @var ExpressionInterface|string
+     */
+    protected $_schema;
+
+    /**
+     * Name of the table
+     *
+     * @var ExpressionInterface|string
+     */
+    protected $_table;
+
+    /**
+     * @inheritDoc
+     *
+     * @param string|ExpressionInterface $schema Name of the schema
+     * @param string|ExpressionInterface $table Name of the table
+     */
+    public function __construct($schema, $table)
+    {
+        $this->_schema = $schema;
+        $this->_table = $table;
+    }
+
+    /**
+     * Converts the Node into a SQL string fragment.
+     *
+     * @param \Cake\Database\ValueBinder $generator Placeholder generator object
+     * @return string
+     */
+    public function sql(ValueBinder $generator)
+    {
+        $schema = $this->_schema;
+        if ($schema instanceof ExpressionInterface) {
+            $schema = $schema->sql($generator);
+        }
+
+        $table = $this->_table;
+        if ($table instanceof ExpressionInterface) {
+            $table = $table->sql($generator);
+        }
+
+        return sprintf('%s.%s', $schema, $table);
```

Comment:
The `quoteIdentifier` setting seems not work. I am not sure that it should be fixed within this method though.
```sql
FROM 
  `users` `Users` 
  LEFT JOIN cakephp2.users `ExternalUsers` ON `Users`.`id` = (`ExternalUsers`.`id`)
```